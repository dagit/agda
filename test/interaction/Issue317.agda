module Issue317 (A : Set) where

postulate F : Set → Set

-- Try evaluating F A at the top-level:
--
-- 1,3-4
-- Not in scope:
--   A at 1,3-4
-- when scope checking A
--
-- OK, in that case the inferred type of F should be
-- (A : Set) → Set → Set, right? No, it isn't, it's Set → Set.
--
-- I think the parameters should be in scope when "top-level" commands
-- are executed, because these commands should behave in the same way
-- as commands executed in a top-level goal at the end of the module.
-- It seems as if the implementation of
-- Agda.Interaction.BasicOps.atTopLevel has to be modified.
