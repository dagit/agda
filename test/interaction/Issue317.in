top_command (cmd_load currentFile [])

-- Should be "Set â†’ Set".
top_command (cmd_infer_toplevel Normalised "F")

-- Should be "F A".
top_command (cmd_compute_toplevel False "F A")
