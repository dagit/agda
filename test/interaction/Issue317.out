(agda2-status-action "")
(agda2-status-action "Checked")
(agda2-info-action "*All Goals*" "" nil)
(agda2-goals-action '())
(agda2-status-action "Checked")
(agda2-info-action "*Inferred Type*" "Set â†’ Set" nil)
(agda2-goals-action '())
(agda2-status-action "Checked")
(agda2-info-action "*Normal Form*" "F A" nil)
(agda2-goals-action '())
